Library.Tests/UnitTests/IssueTests.cs

```diff
@@ -0,0 +1,203 @@
+using Library.Tests.UnitTests;
+using Library.Domain.Models;
+using Xunit;
+
+namespace Library.Tests.UnitTests;
+
+/// <summary>
+/// Unit tests for querying library data using LINQ over in-memory seed collections.
+/// </summary>
+/// <remarks>
+/// Covers the following scenarios:
+/// 1) Issued books ordered by title;
+/// 2) Top-5 readers by number of books in a given period (last 6 months);
+/// 3) Readers who took books for the longest period, ordered by full name;
+/// 4) Top-5 publishers by number of issued books in the last year;
+/// 5) Top-5 least popular books in the last year.
+/// </remarks>
+public class LibraryQueriesTests(DataSeed seed) : IClassFixture<DataSeed>
+{
+    /// <summary>
+    /// Returns distinct issued book titles ordered alphabetically.
+    /// </summary>
+    [Fact]
+    public void IssuedBooks_ShouldBeOrderedByTitle()
+    {
+        // Act
+        var result = seed.Issues
+            .Select(i => i.Book.Title)
+            .Distinct()
+            .OrderBy(t => t)
+            .ToList();
+
+        // Expected: only titles that actually appear in Issues, sorted A→Z
+        var expected = new[]
+        {
+            "A Month in the Country",
+            "Ancient Philosophy",
+            "Children's Tales",
+            "Collected Works",
+            "Crime and Punishment",
+            "Doctor Zhivago",
+            "Eugene Onegin",
+            "Literary Encyclopedia",
+            "Modern Programming",
+            "Modern Review",
+            "The Enchanted Forest",
+            "The Master and Margarita",
+            "War and Peace"
+        };
+
+        // Assert
+        Assert.Equal(expected, result);
+    }
+
+    /// <summary>
+    /// Returns the top 5 readers who read the most books within the last 6 months.
+    /// </summary>
+    [Fact]
+    public void Top5Readers_ByBooksReadInLast6Months()
+    {
+        // Arrange
+        var start = DateTime.Today.AddMonths(-6);
+        var end = DateTime.Today;
+
+        // Act
+        var result = seed.Issues
+            .Where(i => i.IssueDate >= start && i.IssueDate <= end)
+            .GroupBy(i => i.Reader)
+            .Select(g => new { FullName = g.Key.FullName, BooksCount = g.Count() })
```

Comment:
Так как наименование свойства уже FullName, именование можно не делать:
```suggestion
            .Select(g => new { g.Key.FullName, BooksCount = g.Count() })
```